lib/OpenCloud/ObjectStore/Resource/DataObject.php

```diff
@@ -354,6 +382,26 @@ public function delete($params = array())
     {
         return $this->getService()->getClient()->delete($this->getUrl())->send();
     }
+    
+    /**
+     * @param string $source Path (`container/object') of other object to symlink this object to
+     * @return \Guzzle\Http\Message\Response
+     */
+    public function symlink($source)
```

Comment:
I like this a lot. I think there's value in having methods for both directions, given the validations you outlined. Thank you! I will review the new set of changes shortly.